$NetBSD: patch-CMakeLists.txt,v 1.2 2026/02/24 08:32:43 adam Exp $

- Disable precompiled headers with GCC, incompatible with -fPIC.
- Treat FreeBSD like any other pkgsrc platform.

--- CMakeLists.txt.orig	2026-02-20 23:12:29.000000000 +0000
+++ CMakeLists.txt
@@ -4001,7 +4001,7 @@ if(SDL_SHARED)
   set_property(TARGET SDL3-shared PROPERTY INTERFACE_SDL_VERSION "SDL${SDL3_VERSION_MAJOR}")
   set_property(TARGET SDL3-shared APPEND PROPERTY EXPORT_PROPERTIES "SDL_FULL_VERSION")
   set_property(TARGET SDL3-shared PROPERTY SDL_FULL_VERSION "${PROJECT_VERSION}")
-  if(NOT CMAKE_VERSION VERSION_LESS "3.16")
+  if(NOT CMAKE_VERSION VERSION_LESS "3.16" AND NOT USE_GCC)
     target_precompile_headers(SDL3-shared PRIVATE "$<$<OR:$<COMPILE_LANGUAGE:C>,$<COMPILE_LANGUAGE:CXX>>:${PROJECT_SOURCE_DIR}/src/SDL_internal.h>")
   endif()
 endif()
@@ -4027,7 +4027,7 @@ if(SDL_STATIC)
   set_property(TARGET SDL3-static PROPERTY INTERFACE_SDL_VERSION "SDL${SDL3_VERSION_MAJOR}")
   set_property(TARGET SDL3-static APPEND PROPERTY EXPORT_PROPERTIES "SDL_FULL_VERSION")
   set_property(TARGET SDL3-static PROPERTY SDL_FULL_VERSION "${PROJECT_VERSION}")
-  if(NOT CMAKE_VERSION VERSION_LESS "3.16")
+  if(NOT CMAKE_VERSION VERSION_LESS "3.16" AND NOT USE_GCC)
     target_precompile_headers(SDL3-static PRIVATE "$<$<OR:$<COMPILE_LANGUAGE:C>,$<COMPILE_LANGUAGE:CXX>>:${PROJECT_SOURCE_DIR}/src/SDL_internal.h>")
   endif()
 endif()
@@ -4072,7 +4072,7 @@ else()
 endif()
 set(SDL_INSTALL_CMAKEDIR_ROOT "${SDL_INSTALL_CMAKEDIR_ROOT_DEFAULT}" CACHE STRING "Root folder where to install SDL3Config.cmake related files (SDL3 subfolder for MSVC projects)")
 
-if(FREEBSD)
+if(FALSE)
   # FreeBSD uses ${PREFIX}/libdata/pkgconfig
   set(SDL_PKGCONFIG_INSTALLDIR "libdata/pkgconfig")
 else()
